daemon/mount.c

```diff
@@ -326,13 +327,29 @@ mount_private_tmp(void)
 		ERROR_ERRNO("Could not unshare host mount ns!");
 		return -1;
 	}
-	if (umount("/tmp") < 0 && errno != ENOENT) {
-		ERROR_ERRNO("Could not umount /tmp");
+	if (file_is_mountpoint("/tmp")) {
+		if (umount("/tmp") < 0 && errno != ENOENT) {
+			ERROR_ERRNO("Could not umount /tmp");
+			return -1;
+		}
+	}
+	if (mount(NULL, "/", NULL, MS_REC | MS_PRIVATE, NULL) < 0) {
+		ERROR_ERRNO("Could not mount / MS_PRIVATE");
 		return -1;
 	}
-	if (mount("tmpfs", "/tmp", "tmpfs", MS_RELATIME | MS_NOSUID | MS_NODEV, NULL) < 0) {
+	if (file_is_mountpoint("/sys")) {
+		if (mount(NULL, "/sys", NULL, MS_REC | MS_PRIVATE, NULL) < 0) {
```

Comment:
For consistency, we should also set /proc and /dev (if feature_devtpmfs is used) to propagation type MS_PRIVATE

Reply:
done